samples/EverythingServer/Program.cs

```diff
@@ -0,0 +1,201 @@
+﻿using Microsoft.Extensions.Hosting;
+using ModelContextProtocol.Protocol.Types;
+using EverythingServer;
+using ModelContextProtocol.Server;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.DependencyInjection;
+using EverythingServer.Prompts;
+using EverythingServer.Tools;
+using ModelContextProtocol.Protocol.Messages;
+
+var builder = Host.CreateEmptyApplicationBuilder(settings: null);
```

Comment:
```suggestion
var builder = Host.CreateApplicationBuilder(args);
builder.Logging.AddConsole(consoleLogOptions =>
{
    // Configure all logs to go to stderr
    consoleLogOptions.LogToStandardErrorThreshold = LogLevel.Trace;
});
```
We updated our hosted stdio samples to use stderr for logging in #173

Reply:
@halter73 making this change causes a problem as the .NET host writes a bunch of stuff to `stdout`, which causes a client to (potentially) error:

```
$> dotnet run
info: ModelContextProtocol.Protocol.Transport.StdioServerTransport[728055718]
      Transport entering read messages loop for Server (stream) (EverythingServer)
info: ModelContextProtocol.Protocol.Transport.StdioServerTransport[625848173]
      Transport waiting for message for Server (stream) (EverythingServer)
info: ModelContextProtocol.Protocol.Transport.StdioServerTransport[671549983]
      Transport sending message for Server (stream) (EverythingServer) with ID (no id), JSON (null)
{"jsonrpc":"2.0","method":"notifications/message","params":{"level":"alert","data":"Alert-level message"}}
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: D:\mcp-csharp-sdk\samples\EverythingServer
```

This is the console logs when connecting using MCP Inspector:

![image](https://github.com/user-attachments/assets/ad403dc4-552e-4fb2-bc72-b80c3cac9859)

Reply:
Given that the current PR is still using CreateApplicationBuilder with LogToStandardErrorThreshold = LogLevel.Trace, I assume you figured this out. The host should only write to stderr in this configuration. Were you accidently testing with an old build or something?

Reply:
that was merged with `main` (at the time of the comment) and using the change you suggested above. Is there something I should change it to?

Reply:
Nope. What you have now in the PR looks good. I think we were talking past each other. I thought you were saying what's in the PR was broken for you.